### PR TITLE
Analysis tweaks

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
NO

#### Description
Ensure all code analyzers run so that we can control suppression with editor config. 

Also fixes most of the performance related code quality analyzer failures. Let me know if you don't agree with any of these. 

#### Todos
- [ ] Test

### CODE REVIEW

File: src/NzbDrone.Common/OAuth/WebParameterCollection.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class WebParameterCollection : IList<WebParameter>
             {
                 var parameters = this.Where(p => p.Name.Equals(name));
 
-                if (parameters.Count() == 0)
+                if (!parameters.Any())
```
Comment (markus101) [inline src/NzbDrone.Common/OAuth/WebParameterCollection.cs:19]: Could use `.Empty()` extension

File: src/NzbDrone.Common/Http/HttpUri.cs [hunk 2]
```diff
@@ -170,7 +170,7 @@ private static string CombineRelativePath(string basePath, string relativePath)
 
             if (baseSlashIndex >= 0)
             {
-                return basePath.Substring(0, baseSlashIndex) + "/" + relativePath;
+                return string.Concat(basePath.AsSpan(0, baseSlashIndex), "/", relativePath);
```
Comment (markus101) [inline src/NzbDrone.Common/Http/HttpUri.cs:173]: String interpolation instead?

File: src/NzbDrone.Common/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -131,7 +131,7 @@ public static bool ContainsIgnoreCase(this string text, string contains)
 
         public static string WrapInQuotes(this string text)
         {
-            if (!text.Contains(" "))
+            if (!text.Contains(' '))
```
Comment (markus101) [inline src/NzbDrone.Common/Extensions/StringExtensions.cs:134]: Single quotes feels weird here, even though it's just a char, is there an advantage to it?

Comment (Qstick) [reply src/NzbDrone.Common/Extensions/StringExtensions.cs:134]: From Microsoft rule docs:

When searching for a single character, using string.Contains(char) offers better performance than string.Contains(string).

File: src/NzbDrone.Common/OAuth/WebParameterCollection.cs (hunk 1 above)
Comment (Qstick) [reply src/NzbDrone.Common/OAuth/WebParameterCollection.cs:19]: Yep, I keep forgetting that one 😂

**Comment by Qstick** [comment]
Tweaked based on comments, build on TC isn't respecting rules it seems. Something is funny there

**Comment by Qstick** [comment]
This should be fixed now, just failing the unit tests that are bad on dev

**Comment by markus101** [comment]
Doh, fixed the tests.

**Review by markus101 — APPROVED** [review]
Looks good.

**Pull request merged**